Go/Go.sublime-syntax

```diff
@@ -432,28 +432,69 @@ contexts:
     # Dot without integer
     - match: \.\d+(?:{{exponent}}\d+)?
       scope: invalid.deprecated.go
+    # Hex float literal, no fraction
+    - match: (0[xX])(?:_)?\h+(?:_\h+)*([pP])([\+\-])?\d+(?:_\d+)*(i)?
+      scope: constant.numeric.float.go
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.go
+        2: punctuation.section.exponent.go
+        3: keyword.operator.go
+        4: storage.type.numeric.imaginary.go
+    # Hex float literal, with fraction
+    - match: (0[xX])(?:_)?\h*(?:_\h+)*(\.)\h*(?:_\h+)*([pP])([\+\-])?\d+(?:_\d+)*(i)?
+      scope: constant.numeric.float.go
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.go
+        2: punctuation.separator.decimal.go
+        3: punctuation.section.exponent.go
+        4: keyword.operator.go
+        5: storage.type.numeric.imaginary.go
+    # Hex float literal, no fraction, missing exponent digits
+    - match: (0[xX])\h+([pP])([\+\-])?
+      scope: invalid.illegal.go
+    # Hex float literal, with fraction, missing exponent digits
+    - match: (0[xX])\h*(\.)\h*([pP])([\+\-])?
+      scope: invalid.illegal.go
```

Comment:
We try not to get in the way of the user with attention-seeking highlighting of invalid syntax that easily occurs while typing. I believe these would be a good example of that. If there was no highlighting for these at all, it should be enough to signal that these aren't proper numeric constants.

Reply:
Sounds good. I noticed a couple of `invalid.[illegal|deprecated]` in there and probably went a little overboard adding new ones. I removed all the `illegal.*` scopes I added.